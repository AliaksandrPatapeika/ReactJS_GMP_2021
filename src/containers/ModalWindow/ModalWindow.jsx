import './ModalWindow.less';

import React, {Component} from 'react';

import Blur from '../../components/Blur';
import Button from '../../components/Button';
import Logo from '../../components/Logo';
import ModalWindowContext from '../../context';
import AddMovie from './forms/AddMovie';
import DeleteMovie from './forms/DeleteMovie';
import EditMovie from './forms/EditMovie';

class ModalWindow extends Component {
  componentDidUpdate() {
    const {activeModalWindow} = this.context;

    if (activeModalWindow) {
      window.addEventListener('keydown', this.handleKeyDown);
    } else {
      window.removeEventListener('keydown', this.handleKeyDown);
    }
  }

  componentWillUnmount() {
    window.removeEventListener('keydown', this.handleKeyDown);
  }

	handleKeyDown = (event) => {
	  const {closeModalWindow} = this.context;

	  if (event.key === 'Escape') {
	    closeModalWindow();
	  }
	}

	render() {
	  const {activeModalWindow, activeMovie, closeModalWindow} = this.context;

	  return (
  <>
    {activeModalWindow && (
    <div className="modalWindowContainer">
      <Blur>
        <Logo />
        <div className="modalWindow">
          <Button className="closeButton" onClick={closeModalWindow}>
            <i className="fa fa-times" aria-hidden="true" />
          </Button>
          <div className="modalBody">
            {activeModalWindow === 'addMovie' && (
            <AddMovie title="ADD MOVIE" />
            )}
            {activeModalWindow === 'editMovie' && (
            <EditMovie title="EDIT MOVIE" movie={activeMovie} />
            )}
            {activeModalWindow === 'deleteMovie' && (
            <DeleteMovie title="DELETE MOVIE" movie={activeMovie} />
            )}
          </div>
        </div>
      </Blur>
    </div>
    )}
  </>
	  );
	}
}

ModalWindow.contextType = ModalWindowContext;

export default ModalWindow;
