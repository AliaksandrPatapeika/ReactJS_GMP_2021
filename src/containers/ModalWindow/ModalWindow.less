@import "../../less/variables";

.modalWindowContainer {
	background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url('../../img/bgImage.jpg');
	background-size: cover;

	.logoContainer {
		padding: 22px 62px 0;
	}

	.modalWindow {
		display: flex;
		flex-direction: column;
		margin: 80px auto;
		border-radius: 6px;
		background: @black;
		max-width: 726px;

		.closeButton {
			margin: 20px 20px 0 0;
			font-size: 30px;
		}

		.modalBody {
			display: flex;
			flex-direction: column;
			padding: 0 88px;

			.title {
				color: @white;
				font-size: 44px;
				font-family: @NetflixSans-Thin;
				letter-spacing: 2px;
			}

			.formInput {
				display: flex;
				flex-direction: column;
			}

			.content {
				margin-top: 24px;
				color: @white;
				font-size: 25px;
				font-family: @NetflixSans-Thin;
			}

			.label {
				margin: 30px 0 10px 0;
				color: @red;
				font-size: 17px;
				font-family: @NetflixSans-Regular;
			}

			.modalFooter {
				display: flex;
				justify-content: flex-end;
				margin: 50px 0 73px 0;

				.button {
					cursor: pointer;
					width: 190px;
					padding: 15px 0;
					border: 2px solid @red;
					border-radius: 4px;
					color: @red;
					font-size: 22px;
					font-family: @NetflixSans-Regular;
					background-color: transparent;

					&:hover {
						color: @red-hover;
						border-color: @red-hover;
					}

					&:last-of-type {
						margin-left: 15px;
						color: @white;
						background-color: @red;

						&:hover {
							background-color: @red-hover;
						}
					}
				}
			}
		}
	}
}

/* Replacing the calendar icon + placeholder for input-type value */
input[type="date"] {
	position: relative;

	&::-webkit-calendar-picker-indicator {
		cursor: pointer;
		color: transparent;
		background: none;
		z-index: 1;
		outline: none;
	}

	&:not(:focus):not(:valid)::-webkit-datetime-edit-year-field,
	&:not(:focus):not(:valid)::-webkit-datetime-edit-month-field,
	&:not(:focus):not(:valid)::-webkit-datetime-edit-day-field,
	&:not(:focus):not(:valid)::-webkit-datetime-edit-text {
		color: transparent;
	}

	&:after {
		content: attr(placeholder);
		position: absolute;
		/* Placeholder color */
		color: @placeholder;
	}

	&:focus:after,
	&:valid:after {
		content: '';
	}

	&:before {
		content: '\f073';
		position: absolute;
		right: 15px;
		/* This is the calendar icon in FontAwesome */
		display: block;
		color: @red;
		font-size: 24px;
		font-family: @FontAwesome;
		background: none;
	}

	&:hover:before {
		color: @red-hover;
	}
}