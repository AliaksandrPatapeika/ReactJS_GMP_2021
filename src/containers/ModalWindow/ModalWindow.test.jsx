import React from 'react';
import renderer from 'react-test-renderer';

import ModalWindow from './ModalWindow';

describe('ModalWindow', () => {
  test('renders correctly', () => {
    const modalWindow = renderer
      .create(<ModalWindow />)
      .toJSON();

    expect(modalWindow).toMatchSnapshot();
  });
});
