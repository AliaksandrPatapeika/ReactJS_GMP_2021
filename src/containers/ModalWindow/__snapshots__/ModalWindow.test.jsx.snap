// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWindow renders correctly 1`] = `null`;
